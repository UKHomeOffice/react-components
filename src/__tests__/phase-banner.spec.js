import React from 'react'
import { mount, render } from 'enzyme'
import PhaseBanner from 'govuk-react-components/phase-banner';

describe('The Phase Banner Component', () => {

  it('should render a phase of "prototype" by default', () => {
    expect(
      render(<PhaseBanner />)
    ).toMatchSnapshot()
  });

  it('should render the phase when provided', () => {
    expect(
      render(<PhaseBanner phase="beta" />)
    ).toMatchSnapshot()
  });

  it('should contain any children that are passed', () => {
    expect(
      render(<PhaseBanner phase="beta"><span>My own custom text</span></PhaseBanner>)
    ).toMatchSnapshot()
  });

  it('should contain default text and link if a feedback url is provided', () => {
    expect(
      render(<PhaseBanner phase="beta" feedbackUrl="http://example.com" />)
    ).toMatchSnapshot()
  });

  it('should be empty if no feedbackUrl or children are provided', () => {
    expect(
      render(<PhaseBanner phase="beta" />)
    ).toMatchSnapshot()
  });

})
