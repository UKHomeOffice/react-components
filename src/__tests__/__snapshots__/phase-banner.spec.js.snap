// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Phase Banner Component should be empty if no feedbackUrl or children are provided 1`] = `
<div
  class="phase-banner"
>
  <p>
    <strong
      class="phase-tag"
    >
      beta
    </strong>
  </p>
</div>
`;

exports[`The Phase Banner Component should contain any children that are passed 1`] = `
<div
  class="phase-banner"
>
  <p>
    <strong
      class="phase-tag"
    >
      beta
    </strong>
    <span>
      My own custom text
    </span>
  </p>
</div>
`;

exports[`The Phase Banner Component should contain default text and link if a feedback url is provided 1`] = `
<div
  class="phase-banner"
>
  <p>
    <strong
      class="phase-tag"
    >
      beta
    </strong>
    <span>
      This is a new service â€“ your 
      <a
        href="http://example.com"
      >
        feedback
      </a>
       will help us to improve it.
    </span>
  </p>
</div>
`;

exports[`The Phase Banner Component should render a phase of "prototype" by default 1`] = `
<div
  class="phase-banner"
>
  <p>
    <strong
      class="phase-tag"
    >
      prototype
    </strong>
  </p>
</div>
`;

exports[`The Phase Banner Component should render the phase when provided 1`] = `
<div
  class="phase-banner"
>
  <p>
    <strong
      class="phase-tag"
    >
      beta
    </strong>
  </p>
</div>
`;
